node/src/blockchain/blockchain_bridge.rs

```diff
@@ -213,60 +211,35 @@ impl BlockchainBridge {
         }
     }
 
-    pub fn make_connections(
-        blockchain_service_url: Option<String>,
-        data_directory: PathBuf,
-        chain: Chain,
-    ) -> (
-        Box<dyn BlockchainInterface>,
-        Box<dyn PersistentConfiguration>,
-    ) {
-        let blockchain_interface: Box<dyn BlockchainInterface> = {
-            match blockchain_service_url {
-                Some(url) => match Http::new(&url) {
-                    Ok((event_loop_handle, transport)) => {
-                        Self::construct_blockchain_interface(event_loop_handle, transport, chain)
-                    }
-                    Err(e) => panic!("Invalid blockchain node URL: {:?}", e),
-                },
-                None => Box::new(BlockchainInterfaceClandestine::new(chain)),
-            }
-        };
+    pub fn initialize_persistent_configuration(
+        data_directory: &Path,
+    ) -> Box<dyn PersistentConfiguration> {
         let config_dao = Box::new(ConfigDaoReal::new(
             DbInitializerReal::default()
-                .initialize(
-                    &data_directory,
-                    DbInitializationConfig::panic_on_migration(),
-                )
-                .unwrap_or_else(|err| db_connection_launch_panic(err, &data_directory)),
+                .initialize(data_directory, DbInitializationConfig::panic_on_migration())
+                .unwrap_or_else(|err| db_connection_launch_panic(err, data_directory)),
         ));
-        (
-            blockchain_interface,
-            Box::new(PersistentConfigurationReal::new(config_dao)),
-        )
+        Box::new(PersistentConfigurationReal::new(config_dao))
     }
 
-    // TODO when we have multiple chains of fundamentally different architectures and also ability to switch them,
-    // this should probably be replaced by a HashMap of a distinct blockchain interface for each chain
-    fn construct_blockchain_interface(
-        http_event_loop_handle: EventLoopHandle,
-        transport: Http,
+    pub fn initialize_blockchain_interface(
+        blockchain_service_url_opt: Option<String>,
         chain: Chain,
     ) -> Box<dyn BlockchainInterface> {
-        let gas_limit_const_part = web3_gas_limit_const_part(chain);
-        Box::new(BlockchainInterfaceNonClandestine::new(
-            transport,
-            http_event_loop_handle,
-            chain,
-            gas_limit_const_part,
-        ))
+        match blockchain_service_url_opt {
+            Some(url) => {
+                let initializer = BlockchainInterfaceInitializer {};
```

Comment:
It looks to me as though this should be a collaborator that's part of `BlockchainBridge`, and that this function should be a method instead. In that case, we could make `BlockchainInterfaceInitializer` smarter and mock it.